src/utils/src5.ts

```diff
@@ -0,0 +1,36 @@
+import { RpcProvider } from '../provider';
+import type { BigNumberish } from '../types';
+import { toHex } from './num';
+
+/**
+ * Implementation of ERC165 introspection.
+ * Verify if a contract has implemented some standard functionalities.
+ * @param {RpcProvider} provider the provider to access to Starknet.
+ * @param {BigNumberish} contractAddress the address of the contract to check.
+ * @param {BigNumberish} interfaceId the hash of the functionally to check.
+ * @returns {boolean} true if the interfaceId is implement in this contract.
+ * @example
+ * ```typescript
+ * const snip9InterfaceV2Id = constants.SNIP9_V2_INTERFACE_ID;
+ * const result = src5.supportInterface(myProvider, accountContractAddress, snip9InterfaceV2Id);
+ * // result = true
+ * ```
+ */
+export async function supportsInterface(
+  provider: RpcProvider,
+  contractAddress: BigNumberish,
+  interfaceId: BigNumberish
+): Promise<boolean> {
+  const call = {
+    contractAddress: toHex(contractAddress),
+    entrypoint: 'supports_interface',
+    calldata: [toHex(interfaceId)],
+  };
+  try {
+    const resp = await provider.callContract(call);
+    return BigInt(resp[0]) !== 0n;
+  } catch {
+    // account not compatible with ERC165 (introspection)
+    return false;
```

Comment:
I think it's correct.
I return false in 2 cases :
1. ERC165 compatible & SNIP-9 not compatible (ex : OpenZeppelin).
2. ERC165 not compatible, so can't be SNIP-9 compatible (ex: minimalistic account contracts).
And I don't want to throw an error in case 2 ; The response means  that the account is `not compatible with SNIP-9`, and the user has to handle this case exactly as case 1.
Looks good for me.